float64 weight
---
bool success